dataframely/_serialization.py

```diff
@@ -118,3 +121,367 @@ def object_hook(self, dct: dict[str, Any]) -> Any:
+                in the collection metadata, but it is passed separately here to ensure that
+                each member can also be read back as an individual frame.
+        """
+
+    @abstractmethod
+    def scan_collection(
+        self, members: Iterable[str], **kwargs: Any
+    ) -> tuple[dict[str, pl.LazyFrame], list[SerializedCollection | None]]:
+        """Lazily read  all collection members from the storage backend.
+
+        Args:
+            members: Collection member names to read.
+            kwargs: Additional keyword arguments to pass to the underlying storage.
+                Refer to the individual implementation to see which keywords are available.
+        Returns:
+            A tuple of the collection data and metadata if available.
+            Depending on the storage implementation, multiple copies of the metadata
+            may be available, which are returned as a list.
+            It is up to the caller to decide how to handle the presence/absence/consistency
+            of the returned values.
+        """
+
+    @abstractmethod
+    def read_collection(
+        self, members: Iterable[str], **kwargs: Any
+    ) -> tuple[dict[str, pl.LazyFrame], list[SerializedCollection | None]]:
+        """Lazily read  all collection members from the storage backend.
+
+        Args:
+            members: Collection member names to read.
+            kwargs: Additional keyword arguments to pass to the underlying storage.
+                Refer to the individual implementation to see which keywords are available.
+        Returns:
+            A tuple of the collection data and metadata if available.
+            Depending on the storage implementation, multiple copies of the metadata
+            may be available, which are returned as a list.
+            It is up to the caller to decide how to handle the presence/absence/consistency
+            of the returned values.
+        """
+
+
+class ParquetStorageBackend(StorageBackend):
+    """IO manager that stores data and metadata in parquet files on a file system.
+
+    Single frames are stored as individual parquet files
+
+    Collections are stored as directories.
+    """
+
+    # ----------------------------------- Schemas -------------------------------------
+    def sink_frame(
+        self, lf: pl.LazyFrame, serialized_schema: SerializedSchema, **kwargs: Any
+    ) -> None:
+        """This method stores frames as individual parquet files.
+
+        Args:
+            lf: LazyFrame containing the data to be stored.
+            kwargs: The "file" kwarg is required to specify where data is stored.
+                It should point to a parquet file. If hive partitioning is used,
+                it should point to a directory.
+                The "metadata" kwarg is supported to pass a dictionary of parquet
+                metadata.
+                Additional keyword arguments are passed to polars.
```

Comment:
I'm not sure if all this documentation is required, it should be self-explanatory from the superclass. I would even do away with the summary line here and simply expect maintainers (this is not user-facing) to read the class docs (or the little code that's in here).